src/Simulation/Simulators/OpenSystemsSimulator/DataModel/ArrayConverter.cs

```diff
@@ -3,249 +3,245 @@
+///     compatible with the representation used by Rust's <c>ndarray</c>
+///     and <c>serde</c> crates.
+///   </para>
+///   <para>
+///     In particular, each JSON object produced by this converter has
+///     three properties:
+///   </para>
+///   <list type="bullet">
+///     <item><c>"v"</c>: Indicates the <c>ndarray</c> format version being
+///       being used. Always <c>1</c> for this implementation.</item>
+///     <item><c>"dim"</c>: Lists the dimensions of the array being
+///       serialized. Will always contain one less dimension than the
+///       original array, as one dimension represents the real and imaginary
+///       components of each element.</item>
+///     <item><c>"data"</c>: Lists the elements of the array, with the
+///       right-most dimension varying the fastest (defined by <c>ndarray</c>
+///       as "logical order", and by NumPy as "C-ordered").</item>
+///   </list>
+/// </remarks>
+public class ComplexArrayConverter : JsonConverter<NDArray>
 {
-    /// <summary>
-    ///     Converts <see cref="NDArray" /> instances of dtype <c>double</c>
-    ///     and with a trailing index of length 2 to and from a JSON
-    ///     serialization.
-    /// </summary>
-    /// <remarks>
-    ///   <para>    
-    ///     The JSON representation consumed and produced by this converter is
-    ///     compatible with the representation used by Rust's <c>ndarray</c>
-    ///     and <c>serde</c> crates.
-    ///   </para>
-    ///   <para>
-    ///     In particular, each JSON object produced by this converter has
-    ///     three properties:
-    ///   </para>
-    ///   <list type="bullet">
-    ///     <item><c>"v"</c>: Indicates the <c>ndarray</c> format version being
-    ///       being used. Always <c>1</c> for this implementation.</item>
-    ///     <item><c>"dim"</c>: Lists the dimensions of the array being
-    ///       serialized. Will always contain one less dimension than the
-    ///       original array, as one dimension represents the real and imaginary
-    ///       components of each element.
-    ///     <item><c>"data"</c>: Lists the elements of the array, with the
-    ///       right-most dimension varying the fastest (defined by <c>ndarray</c>
-    ///       as "logical order", and by NumPy as "C-ordered").
-    ///   </list>
-    /// </remarks>
-    public class ComplexArrayConverter : JsonConverter<NDArray>
+    public override NDArray Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
     {
-        public override NDArray Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        // By the end, we need to have read "v", "dim", and "data".
+        int? version = null;
+        List<int>? dims = null;
+        List<Complex>? data = null;
+
+        // We require that the reader be in the right state to read an
+        // object.
+        if (reader.TokenType != JsonTokenType.StartObject)
```

Comment:
If the `if` (in line 51) is entered then the definitions above (in lines 45 - 47) are useless (never used). Consider moving the code in lines 45 - 47 to after the line 54 (if C# supports that).